### PR TITLE
Remove dependency on mysqlclient

### PR DESCRIPTION
**Pull request by mss**
mysqlclient is just one possible backend for SQLAlchemy and requires various native build stuff to be installed. Since it is possible to use automua with PyMySQL or completely different databases as well it shouldn't be a hard dependency.

### CODE REVIEW

**Comment by HLFH** [comment]
Hi,

Thanks!
I'll look into it ASAP.

File: pyproject.toml [hunk 1]
```diff
@@ -44,7 +44,6 @@ dependencies =[
     'SQLAlchemy==2.0.0b3',
     'alembic==1.8.1',
     'lxml==4.9.1',
```
Comment (HLFH) [inline pyproject.toml:46]: I'd remove the trailing comma

File: pyproject.toml [hunk 2]
```diff
@@ -73,4 +72,4 @@ path = "automua/__init__.py"
       "/tests",
       "/contrib/flask.sh",
       "/contrib/setupvenv.sh"
-]
\ No newline at end of file
+]
```
Comment (HLFH) [inline pyproject.toml:75]: Not sure what has been done here.

File: requirements-ci.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
```
Comment (HLFH) [inline requirements-ci.txt:3]: Why renaming the default content of the autogenerated files of [`pip-tools`](https://github.com/jazzband/pip-tools)?

File: requirements-test.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
```
Comment (HLFH) [inline requirements-test.txt:3]: Why renaming the default content of the autogenerated files of [`pip-tools`](https://github.com/jazzband/pip-tools)?

File: requirements.txt [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
```
Comment (HLFH) [inline requirements.txt:3]: Why renaming the default content of the autogenerated files of [`pip-tools`](https://github.com/jazzband/pip-tools)?

**Review by HLFH — CHANGES_REQUESTED** [review]
Only the change in `pyproject.toml` could be approved and without the trailing comma.
And the *.txt files need to be updated with `pip-tools`, not manually.

**Comment by HLFH** [comment]
@mss Thanks for your suggestion to remove `mysqlclient` as dep!

The 2023.0 release has been [submitted on Pypi](https://pypi.org/project/automua/), [on Arch Linux](https://aur.archlinux.org/packages/automua) and on GitHub. It has removed `mysqlclient` as a dependency. I have also updated automatically the *.txt files with `pip-compile`, updated the resolver to `backtracking`, and updated the deps.

**Comment by HLFH** [comment]
`mysqlclient` has been added as optdepend. https://github.com/HLFH/automua/commit/d80527d81cc2aabddca99868e899fe4afb2f69fc

File: pyproject.toml (hunk 2 above)
Comment (mss) [reply pyproject.toml:75]: I think my editor added a newline at the end of the file.

**Comment by mss** [comment]
@HLFH thanks for applying this!

Regarding your comments to the `requirements.txt` files: I actually used `pip-compile` to generate them. I guess I used a different version than yours which changed those comments.

**Pull request closed without merging**